xcm/xcm-emulator/example/src/lib.rs

```diff
@@ -0,0 +1,448 @@
+	use sp_runtime::traits::AccountIdConversion;
+	use xcm::{latest::Error::BadOrigin, v3::prelude::*, VersionedMultiLocation, VersionedXcm};
+	use xcm_emulator::TestExt;
+
+	#[test]
+	fn dmp() {
+		Network::reset();
+
+		let remark = test_runtime::RuntimeCall::System(
+			frame_system::Call::<test_runtime::Runtime>::remark_with_event {
+				remark: "Hello from Kusama!".as_bytes().to_vec(),
+			},
+		);
+		KusamaRelay::execute_with(|| {
+			assert_ok!(kusama_runtime::XcmPallet::force_default_xcm_version(
+				kusama_runtime::RuntimeOrigin::root(),
+				Some(3)
+			));
+			assert_ok!(kusama_runtime::XcmPallet::send_xcm(
+				Here,
+				Parachain(1),
+				Xcm(vec![Transact {
+					origin_kind: OriginKind::SovereignAccount,
+					require_weight_at_most: Weight::from_parts(INITIAL_BALANCE as u64, 1024 * 1024),
+					call: remark.encode().into(),
+				}]),
+			));
+		});
+
+		ParachainA::execute_with(|| {
+			use test_runtime::{RuntimeEvent, System};
+			System::events().iter().for_each(|r| println!(">>> {:?}", r.event));
+
+			assert!(System::events().iter().any(|r| matches!(
+				r.event,
+				RuntimeEvent::System(frame_system::Event::Remarked { sender: _, hash: _ })
+			)));
+		});
+	}
+
+	#[test]
+	fn ump() {
+		Network::reset();
+
+		KusamaRelay::execute_with(|| {
+			assert_ok!(kusama_runtime::XcmPallet::force_default_xcm_version(
+				kusama_runtime::RuntimeOrigin::root(),
+				Some(3)
+			));
+			let _ = kusama_runtime::Balances::deposit_creating(
+				&ParaId::from(1).into_account_truncating(),
+				1_000_000_000_000,
+			);
+		});
+
+		let remark = kusama_runtime::RuntimeCall::System(frame_system::Call::<
+			kusama_runtime::Runtime,
+		>::remark_with_event {
+			remark: "Hello from ParachainA!".as_bytes().to_vec(),
+		});
+		ParachainA::execute_with(|| {
+			assert_ok!(test_runtime::PolkadotXcm::force_default_xcm_version(
+				test_runtime::RuntimeOrigin::root(),
+				Some(3)
+			));
+			assert_ok!(test_runtime::PolkadotXcm::send_xcm(
+				Here,
+				Parent,
+				Xcm(vec![
+					UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+					Transact {
+						origin_kind: OriginKind::SovereignAccount,
+						require_weight_at_most: Weight::from_parts(
+							INITIAL_BALANCE as u64,
+							1024 * 1024
+						),
+						call: remark.encode().into(),
+					}
+				]),
+			));
+		});
+
+		KusamaRelay::execute_with(|| {
+			use kusama_runtime::{RuntimeEvent, System};
+			// TODO: https://github.com/paritytech/polkadot/pull/6824 or change this call to
+			// force_create_assets like we do in cumulus integration tests.
+			// assert!(System::events().iter().any(|r| matches!(
+			// 	r.event,
+			// 	RuntimeEvent::System(frame_system::Event::Remarked { sender: _, hash: _ })
+			// )));
+			assert!(System::events().iter().any(|r| matches!(
+				r.event,
+				RuntimeEvent::Ump(polkadot_runtime_parachains::ump::Event::ExecutedUpward(
+					_,
+					Outcome::Incomplete(_, XcmError::NoPermission)
+				))
+			)));
```

Comment:
This seems just to be testing `SafeCallFilter`? What is the value of "testing" a `remark_with_event`?

Reply:
Yes, it's currently hitting the `SafeCallFilter`, but once that's removed - this could be reinstated. Since the purpose of those example tests is to demonstrate common use-cases, not necessarily focusing on the outcome - it could be ok to keep it as is.

According to the discussion [here](https://github.com/paritytech/polkadot/pull/6824#issuecomment-1464913946), as far as I understand it, there's no real reason why the `remark_with_event` call is disabled in the first place, as the actual event that's stored contains a hash and not the original unbounded `Vec`, correct me if I'm wrong.

That said, there's an alternative, which is to `force_create_asset`. It's a bit of a more complicated scenario, but it works.

Reply:
Yes, I understand all that, but this error returns [quite early](https://github.com/paritytech/polkadot/blob/release-v0.9.42/xcm/xcm-executor/src/lib.rs#L550) in the `Transact` instruction. So the only thing this tests is that this `ensure` statement works. As the error indicates, the execution is "incomplete".

Reply:
Right, I'll raise a PR to change that to `force_create_assets`, going to be less confusing. Though it would be nice to see `remark_with_event` whitelisted too, but not as part of this PR.

Reply:
Fixed [here](https://github.com/paritytech/cumulus/pull/2566)